plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,343 @@
+          new_event.runas_user = value
+        # configuration
+        elif counter == 4:
+          new_event.config_name = value
+        # machine
+        elif counter == 5:
+          new_event.machine = value
+        # host application
+        elif counter == 6:
+          new_event.host_application = value
+        # process id
+        elif counter == 7:
+          new_event.process_id = value
+        # version
+        elif counter == 8:
+          new_event.version = value
+        # edition
+        elif counter == 9:
+          new_event.edition = value
+        # compatible version
+        elif counter == 10:
+          new_event.compatible_versions = value
+        # build version
+        elif counter == 11:
+          new_event.build_version = value
+        # clr version
+        elif counter == 12:
+          new_event.clr_version = value
+        # ws man stack version
+        elif counter == 13:
+          new_event.ws_man_stack_version = value
+        # remoting protocol version
+        elif counter == 14:
+          new_event.remoting_protocol_version = value
+        # serialization version
+        elif counter == 15:
+          new_event.serialization_version = value
+        counter += 1
+      # update base event for future use
+      self.base_event = copy.deepcopy(new_event)
+    # transcript with timestamp header
+    elif len(metadata_lines) == 1:
+      if self.base_event is None:
+        parser_mediator.ProduceExtractionWarning('initial transcript seems '
+          'to have missing headers - can not produce event')
+        return False
+      # since we don't have the full headers,
+      # we just copy the headers from the last full event...
+      new_event = copy.deepcopy(self.base_event)
+      if ': ' in metadata_lines[0]:
+        value = metadata_lines[0].split(': ')[1]
+      else:
+        value = None
+      # ... and overwrite the timestamp
+      start_time = self._GetTimestampFromString(value)
+      if start_time is None:
+        parser_mediator.ProduceExtractionWarning('timestamp \"{0!s}\" seems '
+          'invalid - skipping malformed transcript'.format(value))
+        return False
+      start_time.is_local_time = True
+      event_time = time_events.DateTimeValuesEvent(start_time,
+        definitions.TIME_DESCRIPTION_START)
+
+    if transcript_text == '':
+      parser_mediator.ProduceExtractionWarning('skipping transcript {0!s} - '
+          'since transcript text could not be set'.format(key_structure))
+      return False
+    new_event.command = transcript_text
+    # create event if at least the most relevant fields are set
+    if (event_time is not None and new_event.command is not None
+      and new_event.command != ''):
+      parser_mediator.ProduceEventWithEventData(event_time, new_event)
+    else:
+      parser_mediator.ProduceExtractionWarning('skipping transcript {0!s} - '
+        'relevant event values could not be set'.format(key_structure))
```

Comment:
style nit - this line should be indented so it is vertically aligned, but a better layout might be hanging indents.